File: app/src/main/java/ds/vkplus/auth/AccountAuthenticatorService.java
package ds.vkplus.auth;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import static android.accounts.AccountManager.ACTION_AUTHENTICATOR_INTENT;


public class AccountAuthenticatorService extends Service {

	private static AccountAuthenticator AUTHENTICATOR = null;


	@Override
	public IBinder onBind(Intent intent) {
		return intent.getAction().equals(ACTION_AUTHENTICATOR_INTENT) ? getAuthenticator().getIBinder() : null;
	}


	private AccountAuthenticator getAuthenticator() {
		if (AUTHENTICATOR == null)
			AUTHENTICATOR = new AccountAuthenticator(this);
		return AUTHENTICATOR;
	}
}

File: app/src/main/java/ds/vkplus/auth/AccountAuthenticatorService.kt
package ds.vkplus.auth

import android.app.Service
import android.content.Intent
import android.os.IBinder

import android.accounts.AccountManager.ACTION_AUTHENTICATOR_INTENT


public class AccountAuthenticatorService : Service() {

    private val AUTHENTICATOR: AccountAuthenticator by lazy { AccountAuthenticator(this) }

    override fun onBind(intent: Intent): IBinder? {
        return if (intent.action == ACTION_AUTHENTICATOR_INTENT) AUTHENTICATOR.iBinder else null
    }

}

